staging/src/k8s.io/client-go/rest/client.go

```diff
@@ -124,9 +132,16 @@ func NewRESTClient(baseURL *url.URL, versionedAPIPath string, config ContentConf
 		createBackoffMgr: readExpBackoffConfig,
 		Throttle:         throttle,
 		Client:           client,
+		timeout:          timeout,
 	}, nil
 }
 
+// Set Timeout
+func (c *RESTClient) Timeout(timeout time.Duration) *RESTClient {
```

Comment:
This method sets timeout for RESTClient, then how to specify different timeout value for different Verbs?

Reply:
before:
you have to choose verb and can change timeout for that specific verb.
now:
in client you can change timeout and choose the verb according to the need.

I can think of some thing like this.
```go
func RESTClientFor(config *Config) // 
```
in config user can specify different timeout for different verb in map (map[string]time.Duration) then when user choose verb, request will be set with timeout according to the verb

Reply:
you mean to construct every verb a RESTClient?

Reply:
nope I meant
```go
func (c *RESTClient) Verb(verb string) *Request {
	backoff := c.createBackoffMgr()
        timeout := c.timeouts[verb] // something like this
	if c.Client == nil {
		return NewRequest(nil, verb, c.base, c.versionedAPIPath, c.contentConfig, c.serializers, backoff, c.Throttle, timeout)
	}
	return NewRequest(c.Client, verb, c.base, c.versionedAPIPath, c.contentConfig, c.serializers, backoff, c.Throttle, timeout)
}
```

Reply:
@hzxuzhonghu should I go with the above sol?